src/development/add-to-app/ios/add-flutter-screen.md

```diff
@@ -107,13 +143,56 @@ class AppDelegate: FlutterAppDelegate { // More on the FlutterAppDelegate.
 
 ### Show a FlutterViewController with your FlutterEngine
 
+{% samplecode vc %}
+
+{% sample SwiftUI %}
+The following example shows a generic `ContentView` with a
+`Button` hooked to present a [`FlutterViewController`][].
+The `FlutterViewController` uses the `FlutterEngine` instance
+passed in as an `EnvironmentObject`.
+<!--code-excerpt "ContentView.swift" title-->
+```swift
+import SwiftUI
+import Flutter
+
+struct ContentView: View {
+  // Flutter engine is passed in as an EnvironmentObject.
+  @EnvironmentObject var flutterDependencies: FlutterDependencies
+
+  // Button is created to call the showFlutter function when pressed.
+  var body: some View {
+    Button("Show Flutter!") {
+      showFlutter()
+    }
+  }
+
+func showFlutter() {
+    // Get the RootViewController.
+    guard
+      let windowScene = UIApplication.shared.connectedScenes
+        .first(where: { $0.activationState == .foregroundActive && $0 is UIWindowScene }) as? UIWindowScene,
+      let window = windowScene.windows.first(where: \.isKeyWindow),
+      let rootViewController = window.rootViewController
+    else { return }
+
+    // Create the FlutterViewController.
+    let flutterViewController = FlutterViewController(
+      engine: flutterDependencies.npsFlutterEngine,
```

Comment:
IIRC "nps" is from the news feed project, so readers of this doc may not have context.